esx_service/vmdk_ops.py

```diff
@@ -1810,6 +1853,13 @@ def handleVmciRequests(port):
 
         client_socket = c # Bind to avoid race conditions.
 
+        # Check the flag - if set, fail new incoming requests
+        if stopEvent.isSet():
```

Comment:
Can't the clients simply timeout and fail themselves? Thats the error the client will anyway get if the service was successfully stopped (when this loop exited) and the client issued a command after that. This handling of commands during the drain is extra.

Reply:
We can, but I feel if possible, fail the request immediately is still better than timeout, from user experience point of view.

Reply:
@govint - why make them suffer and timeout, when we can return a clear error message right away ? 
It really adds 3 lines to our code, with is about 3 times less than the number of lines we already spent discussing :-)